src/sentry/api/serializers/models/organization_member/utils.py

```diff
@@ -1,32 +1,40 @@
 from collections import defaultdict
-from typing import List, Mapping, Sequence, Set
+from typing import Any, Dict, List, Mapping, Sequence, Set, Tuple, TypeVar
 
 from sentry.api.serializers import serialize
 from sentry.models import OrganizationMember, OrganizationMemberTeam, Team, TeamStatus, User
 
+TeamData = TypeVar("TeamData")
+DictOfMembers = Dict[Any, List[TeamData]]
+MembersTeams = DictOfMembers[str]
+MembersTeamRoles = DictOfMembers[Dict[str, str]]
```

Comment:
Nice.

If I tinkered with this a bit, I might come up with some more aggressive refactoring ideas to make it better to unpack. But this is a great start for clarity.